orderbook/openapi.yml

```diff
@@ -769,3 +810,99 @@ components:
       required:
         - errorType
         - description
+    OrderQuoteSide:
+      description: The buy or sell side when quoting an order.
+      oneOf:
+        - type: object
+          description: Quote a sell order given the final total sell amount including fees
+          properties:
+            kind:
+              type: string
+              enum: [sell]
+            sellAmountBeforeFee:
+              description: |
+                The total amount that is available for the order. From this value, the fee
+                is deducted and the buy amount is calculated.
+              $ref: "#/components/schemas/TokenAmount"
+          required:
+            - kind
+            - sellAmountBeforeFee
+        - type: object
+          description: Quote a sell order given the sell amount.
+          properties:
+            kind:
+              type: string
+              enum: [sell]
+            sellAmountAfterFee:
+              description: The sell amount for the order.
+              $ref: "#/components/schemas/TokenAmount"
+          required:
+            - kind
+            - sellAmountAfterFee
+        - type: object
+          description: Quote a buy order given an exact buy amount.
+          properties:
+            kind:
+              type: string
+              enum: [buy]
+            buyAmountAfterFee:
+              description: The buy amount for the order.
+              $ref: "#/components/schemas/TokenAmount"
+          required:
+            - kind
+            - buyAmountAfterFee
+    OrderQuoteRequest:
+      description: Request fee and price quote.
+      allOf:
+        - $ref: "#/components/schemas/OrderQuoteSide"
+        - type: object
+          properties:
+            sellToken:
+              description: "ERC20 token to be sold"
+              $ref: "#/components/schemas/Address"
+            buyToken:
+              description: "ERC20 token to be bought"
+              $ref: "#/components/schemas/Address"
+            receiver:
```

Comment:
Is the receiver relevant for the price quote?

Reply:
While it may not be needed for the quote, it will be needed to construct the singable order that is meant to be returned from this endpoint.

Reply:
We may offer discounts if the receiver address stakes GP tokens for example (and not just the owner).

Reply:
For the receiver as well as the owner? Wow, ok